### ISSUE TITLE
Keeps forgetting hidden threads

### ISSUE THREAD

**Issue by alidan** [ccd0/4chan-x#1910]
Long story short, I open and hide threads, I keep the threads saved, and I load them up when they 404 to scrape images. 
I have been opening threads with a middle click and shift click macro for each thread I want to keep up with, then closing them out in tab outliner
However I just went to most of the boards I do this to, and they move slow enough that there is no way in hell I only have 4 hidden threads (the threads that I hide based on filters) 

im on  1.13.15.11 of the extension version.

**Comment by alidan** [comment]
Got updated to  1.13.15.12 shortly after this was written
chrome crashed
the extension forgot every single post I made.

**Comment by Corridor89** [comment]
this has been an on-going issue for the last few weeks. see 1875 and 1881
Best thing to try is the beta version of the extension/script

**Comment by alidan** [comment]
Ok on the beta version and on the script rather than extension, someone said that one loses the data less. 
there any major difference between script and extension performance wise?

either way, going to re make all the hidden threads and shit, and see if it remembers it for more then 1 day

**Comment by alidan** [comment]
Ok, Initial threads watched/hidden and everything is remembered. If anyone wants updates reply back in a few days

so far, beta script is doing better than non beta extension.

**Comment by alidan** [comment]
Ok, I watch threads on all the porn boards but yaoi yuri and handsome men 
the 1.14.0.15 version script has forgotten everything after 4 days, which while its not good, is better than 1.13.15.12 given that this one forgot it multiple times a day. 

I use ublock origin and umartix, is it possible that a conflict is happening there? 
also, I don't know how extensions work, is it possible to have an extension lookup a file on your computer, use&write data to it, and just go on with its day? doing this all in browser seems like it has issues.
I know you have import and export settings, would it be possible to have import and export settings for the hide lists? It wouldn't be the ideal solution, but a work around till a more permanent solution is found, along with an extra 'security' measure in case this issue pops up again.

**Comment by Corridor89** [comment]
I don't think an ad blocker would affect it. I have a different ad blocker than you and it still happens. 
And according to the posts in the other threads, the issue has been happening on different browsers. I think they issue would have to be either the script itself of 4chan having some changes that stopped it from working like it did a month ago.

**Comment by alidan** [comment]
well, and I have no idea how anything works but I have an idea. 
my filters work perfectly, there is no issue with them picking up anything
so my guess here is that the hidden thread list works like filters, but instead of a permanent list of shit to block, its a list that culls itself every now and then, I notice when I leave a board open every now and then everything disappears and reappears, i assume that is it checking, because lets say a few hundred threads to check isn't an issue, how about a few thousand, that would likely cause issues. 

so its either something in the 4chanx is causing a full page rather than procedural
web browsers are screwing with what's saved by 4chanx
an extension is screwing with it

I want to rule 4chan out entirely, because if I want to filter 
https://boards.4chan.org/s/thread/18148662
as an example you are going to either filter
1) https://boards.4chan.org/s/thread/18148662
2) */s/thread/18148662
3) */18148662
4) /TV3vOHcSJVlJYmxuTeS0ag==/ + Date
5) /TV3vOHcSJVlJYmxuTeS0ag==/

doing this small test, I filtered the image md5 and reloaded, thread was filtered
I removed the image md5 and reloaded and it was no longer filtered.
This is seems like its the most likely thing to happen, that 4chanx is losing its hide list. 
a save the list to desktop and load list from desktop would be the most likely fix/workaround till the problem is properly addressed.

**Comment by gotmilk0112** [comment]
Even using the beta script, it's still happening. Just happened now. All hidden threads reset, all (You)s removed.

**Comment by ccd0** [comment]
Are you doing anything like clearing cookies periodically?

**Comment by alidan** [comment]
the only cookies I clear out are youtube ones due to the way the great suspender works with it, it will cause an error 400 if left to long, and 4chan more rarely with errors caused by that, but haven't had any in I want to say 5 months with 4chan and even at that, its a very noticeable 'geoscrollpos - *****' that shits everything up and only those get taken out.

**Comment by ccd0** [comment]
> I know you have import and export settings, would it be possible to have import and export settings for the hide lists?

The exported settings includes the hide list.

**Comment by ccd0** [comment]
I found something in the list pruning that might cause a problem if 4chan started sending false 404s. Can you see the issue still happens in v1.14.1.1 (current beta)?

This wouldn't explain watch list additions/removals as in #1881 but that might be a separate issue and possibly already fixed by the other stuff I've done; I'll have to check back on it.

**Comment by alidan** [comment]
Ok, updated and will go through 
/s/ /h/ /e/ /d/ /hr/ /gif/ /aco/ and hide everything older then 5/18/18 12:44am as that is the last time I checked everything.

will be able to get back in I want to say a week and tell you results, possibly sooner if it didn't fix anything.

**Comment by gotmilk0112** [comment]
Nope, I don't touch any of my cookies or browsing data unless I have to.

**Comment by alidan** [comment]
so I hid everything on those boards, 
/s/ reverted
/h/ reverted
/hc/ reverted
/hr/ reverted
I think its not fixed yet.

**Comment by ccd0** [comment]
What do you see when you open
https://a.4cdn.org/s/threads.json
?

**Comment by gotmilk0112** [comment]
https://puu.sh/AoKUJ/2ed02dd9ed.png

That's what it shows for me.

**Comment by alidan** [comment]
[{"page":1,"threads":[{"no":12370429,"last_modified":1453430219},{"no":9112225,"last_modified":1798779601},{"no":18164272,"last_modified":1526708236},{"no":18160360,"last_modified":1526708128},{"no":18167757,"last_modified":1526708118},{"no":18172916,"last_modified":1526708077},{"no":18171271,"last_modified":1526707862},{"no":18174582,"last_modified":1526707847},{"no":18158657,"last_modified":1526707663},{"no":18169657,"last_modified":1526707137},{"no":18156727,"last_modified":1526707118},{"no":18163666,"last_modified":1526706667},{"no":18175187,"last_modified":1526706649},{"no":18169786,"last_modified":1526706176},{"no":18169100,"last_modified":1526706071}]},{"page":2,"threads":[{"no":18176133,"last_modified":1526706069},{"no":18176281,"last_modified":1526707097},{"no":18165315,"last_modified":1526706064},{"no":18165956,"last_modified":1526706060},{"no":18160273,"last_modified":1526705675},{"no":18174521,"last_modified":1526705647},{"no":18169357,"last_modified":1526705587},{"no":18153487,"last_modified":1526705571},{"no":18176303,"last_modified":1526705449},{"no":18167850,"last_modified":1526705366},{"no":18159709,"last_modified":1526705060},{"no":18164752,"last_modified":1526704513},{"no":18163238,"last_modified":1526704501},{"no":18151097,"last_modified":1526703985},{"no":18168542,"last_modified":1526703961}]},{"page":3,"threads":[{"no":18173153,"last_modified":1526703596},{"no":18176193,"last_modified":1526703190},{"no":18173580,"last_modified":1526703100},{"no":18165725,"last_modified":1526703013},{"no":18175770,"last_modified":1526702995},{"no":18166606,"last_modified":1526702656},{"no":18171002,"last_modified":1526702511},{"no":18148974,"last_modified":1526702154},{"no":18147171,"last_modified":1526702011},{"no":18148025,"last_modified":1526701992},{"no":18160159,"last_modified":1526701909},{"no":18149575,"last_modified":1526701636},{"no":18175987,"last_modified":1526701541},{"no":18173213,"last_modified":1526701455},{"no":18171550,"last_modified":1526701440}]},{"page":4,"threads":[{"no":18166139,"last_modified":1526701270},{"no":18176211,"last_modified":1526701101},{"no":18166819,"last_modified":1526701095},{"no":18176198,"last_modified":1526701170},{"no":18175760,"last_modified":1526700392},{"no":18152799,"last_modified":1526700319},{"no":18174612,"last_modified":1526700023},{"no":18139359,"last_modified":1526699723},{"no":18170093,"last_modified":1526699599},{"no":18154585,"last_modified":1526699352},{"no":18158300,"last_modified":1526699239},{"no":18159979,"last_modified":1526698820},{"no":18175213,"last_modified":1526699837},{"no":18165924,"last_modified":1526698552},{"no":18173654,"last_modified":1526698023}]},{"page":5,"threads":[{"no":18172382,"last_modified":1526697958},{"no":18152197,"last_modified":1526697954},{"no":18161542,"last_modified":1526697839},{"no":18160859,"last_modified":1526697520},{"no":18175641,"last_modified":1526697508},{"no":18174939,"last_modified":1526697456},{"no":18156436,"last_modified":1526697218},{"no":18152928,"last_modified":1526696823},{"no":18148662,"last_modified":1526696724},{"no":18168719,"last_modified":1526696446},{"no":18167437,"last_modified":1526696272},{"no":18160791,"last_modified":1526695795},{"no":18171908,"last_modified":1526695765},{"no":18170378,"last_modified":1526695255},{"no":18175660,"last_modified":1526694309}]},{"page":6,"threads":[{"no":18144762,"last_modified":1526694225},{"no":18157306,"last_modified":1526694019},{"no":18169769,"last_modified":1526693873},{"no":18175828,"last_modified":1526693702},{"no":18171782,"last_modified":1526693515},{"no":18176021,"last_modified":1526693313},{"no":18166360,"last_modified":1526692956},{"no":18174075,"last_modified":1526692521},{"no":18146407,"last_modified":1526692328},{"no":18156698,"last_modified":1526690867},{"no":18153989,"last_modified":1526690848},{"no":18170532,"last_modified":1526690842},{"no":18175980,"last_modified":1526690682},{"no":18174208,"last_modified":1526690414},{"no":18173942,"last_modified":1526689848}]},{"page":7,"threads":[{"no":18161468,"last_modified":1526689199},{"no":18175951,"last_modified":1526689039},{"no":18145750,"last_modified":1526688386},{"no":18175944,"last_modified":1526688304},{"no":18158893,"last_modified":1526687981},{"no":18164401,"last_modified":1526687963},{"no":18143725,"last_modified":1526687860},{"no":18167230,"last_modified":1526687855},{"no":18162258,"last_modified":1526687823},{"no":18147784,"last_modified":1526689071},{"no":18170838,"last_modified":1526687099},{"no":18163893,"last_modified":1526687089},{"no":18163404,"last_modified":1526686908},{"no":18169886,"last_modified":1526686545},{"no":18174672,"last_modified":1526686118}]},{"page":8,"threads":[{"no":18174961,"last_modified":1526684430},{"no":18171094,"last_modified":1526683776},{"no":18159920,"last_modified":1526683135},{"no":18165739,"last_modified":1526682510},{"no":18163958,"last_modified":1526681676},{"no":18174018,"last_modified":1526681502},{"no":18147341,"last_modified":1526680946},{"no":18169269,"last_modified":1526680844},{"no":18175591,"last_modified":1526678867},{"no":18141720,"last_modified":1526677881},{"no":18172984,"last_modified":1526677184},{"no":18165813,"last_modified":1526677027},{"no":18175584,"last_modified":1526676957},{"no":18156416,"last_modified":1526676925},{"no":18175567,"last_modified":1526676947}]},{"page":9,"threads":[{"no":18172296,"last_modified":1526675956},{"no":18145282,"last_modified":1526675707},{"no":18175466,"last_modified":1526674012},{"no":18175467,"last_modified":1526678067},{"no":18163868,"last_modified":1526673219},{"no":18169920,"last_modified":1526668353},{"no":18155214,"last_modified":1526668046},{"no":18151845,"last_modified":1526665957},{"no":18164784,"last_modified":1526685130},{"no":18148825,"last_modified":1526663115},{"no":18152717,"last_modified":1526661825},{"no":18172943,"last_modified":1526661816},{"no":18175144,"last_modified":1526706826},{"no":18145901,"last_modified":1526658455},{"no":18155720,"last_modified":1526656459}]},{"page":10,"threads":[{"no":18162507,"last_modified":1526656427},{"no":18173804,"last_modified":1526655945},{"no":18172833,"last_modified":1526655602},{"no":18174867,"last_modified":1526650817},{"no":18174798,"last_modified":1526647354},{"no":18164262,"last_modified":1526640740},{"no":18172854,"last_modified":1526636621},{"no":18144785,"last_modified":1526629012},{"no":18173863,"last_modified":1526628487},{"no":18158339,"last_modified":1526626796},{"no":18119520,"last_modified":1526625926},{"no":18170525,"last_modified":1526624857},{"no":18172114,"last_modified":1526622059},{"no":18145594,"last_modified":1526610650},{"no":18174161,"last_modified":1526665537}]},{"page":11,"threads":[{"no":18173783,"last_modified":1526608623}]}]

that's what I got. keep in mind, I had hidden 9 boards worth of threads,

**Comment by alidan** [comment]
I just went to every board, nothing beyond the normal filters are there.

**Comment by alidan** [comment]
ok went in and re hid everything again, 

[{"page":1,"threads":[{"no":12370429,"last_modified":1453430219},{"no":9112225,"last_modified":1798779601},{"no":18172943,"last_modified":1526711576},{"no":18173580,"last_modified":1526711561},{"no":18171550,"last_modified":1526711551},{"no":18170525,"last_modified":1526711386},{"no":18159979,"last_modified":1526711367},{"no":18166819,"last_modified":1526711361},{"no":18173942,"last_modified":1526711361},{"no":18174582,"last_modified":1526711336},{"no":18156727,"last_modified":1526711296},{"no":18160159,"last_modified":1526711241},{"no":18148662,"last_modified":1526711151},{"no":18173153,"last_modified":1526710516},{"no":18176303,"last_modified":1526710174}]},{"page":2,"threads":[{"no":18159920,"last_modified":1526710172},{"no":18149575,"last_modified":1526710104},{"no":18154585,"last_modified":1526709957},{"no":18163404,"last_modified":1526709452},{"no":18176133,"last_modified":1526709381},{"no":18167757,"last_modified":1526709151},{"no":18165956,"last_modified":1526709091},{"no":18167850,"last_modified":1526708984},{"no":18172854,"last_modified":1526708984},{"no":18163893,"last_modified":1526708720},{"no":18176358,"last_modified":1526709714},{"no":18176193,"last_modified":1526708689},{"no":18163666,"last_modified":1526708681},{"no":18151845,"last_modified":1526708566},{"no":18169657,"last_modified":1526708491}]},{"page":3,"threads":[{"no":18119520,"last_modified":1526708442},{"no":18160360,"last_modified":1526708429},{"no":18164272,"last_modified":1526708236},{"no":18172916,"last_modified":1526708077},{"no":18171271,"last_modified":1526707862},{"no":18158657,"last_modified":1526707663},{"no":18175187,"last_modified":1526706649},{"no":18169786,"last_modified":1526706176},{"no":18169100,"last_modified":1526706071},{"no":18176281,"last_modified":1526707097},{"no":18165315,"last_modified":1526706064},{"no":18160273,"last_modified":1526705675},{"no":18174521,"last_modified":1526705647},{"no":18169357,"last_modified":1526705587},{"no":18153487,"last_modified":1526705571}]},{"page":4,"threads":[{"no":18159709,"last_modified":1526705060},{"no":18164752,"last_modified":1526704513},{"no":18163238,"last_modified":1526704501},{"no":18151097,"last_modified":1526703985},{"no":18168542,"last_modified":1526703961},{"no":18165725,"last_modified":1526703013},{"no":18175770,"last_modified":1526702995},{"no":18166606,"last_modified":1526702656},{"no":18171002,"last_modified":1526702511},{"no":18148974,"last_modified":1526702154},{"no":18147171,"last_modified":1526702011},{"no":18148025,"last_modified":1526701992},{"no":18175987,"last_modified":1526701541},{"no":18173213,"last_modified":1526701455},{"no":18166139,"last_modified":1526701270}]},{"page":5,"threads":[{"no":18176211,"last_modified":1526701101},{"no":18176198,"last_modified":1526701170},{"no":18175760,"last_modified":1526700392},{"no":18152799,"last_modified":1526700319},{"no":18174612,"last_modified":1526700023},{"no":18139359,"last_modified":1526699723},{"no":18170093,"last_modified":1526699599},{"no":18158300,"last_modified":1526699239},{"no":18175213,"last_modified":1526699837},{"no":18165924,"last_modified":1526698552},{"no":18173654,"last_modified":1526698023},{"no":18172382,"last_modified":1526697958},{"no":18152197,"last_modified":1526697954},{"no":18161542,"last_modified":1526697839},{"no":18160859,"last_modified":1526697520}]},{"page":6,"threads":[{"no":18175641,"last_modified":1526697508},{"no":18174939,"last_modified":1526697456},{"no":18156436,"last_modified":1526697218},{"no":18152928,"last_modified":1526696823},{"no":18168719,"last_modified":1526696446},{"no":18167437,"last_modified":1526696272},{"no":18160791,"last_modified":1526695795},{"no":18171908,"last_modified":1526695765},{"no":18170378,"last_modified":1526695255},{"no":18175660,"last_modified":1526694309},{"no":18144762,"last_modified":1526694225},{"no":18157306,"last_modified":1526694019},{"no":18169769,"last_modified":1526693873},{"no":18175828,"last_modified":1526693702},{"no":18171782,"last_modified":1526693515}]},{"page":7,"threads":[{"no":18176021,"last_modified":1526693313},{"no":18166360,"last_modified":1526692956},{"no":18174075,"last_modified":1526692521},{"no":18146407,"last_modified":1526692328},{"no":18156698,"last_modified":1526690867},{"no":18153989,"last_modified":1526690848},{"no":18170532,"last_modified":1526690842},{"no":18175980,"last_modified":1526690682},{"no":18174208,"last_modified":1526690414},{"no":18161468,"last_modified":1526689199},{"no":18175951,"last_modified":1526689039},{"no":18145750,"last_modified":1526688386},{"no":18175944,"last_modified":1526688304},{"no":18158893,"last_modified":1526687981},{"no":18164401,"last_modified":1526687963}]},{"page":8,"threads":[{"no":18143725,"last_modified":1526687860},{"no":18167230,"last_modified":1526687855},{"no":18162258,"last_modified":1526687823},{"no":18147784,"last_modified":1526689071},{"no":18170838,"last_modified":1526687099},{"no":18169886,"last_modified":1526686545},{"no":18174672,"last_modified":1526686118},{"no":18174961,"last_modified":1526684430},{"no":18171094,"last_modified":1526683776},{"no":18165739,"last_modified":1526682510},{"no":18163958,"last_modified":1526681676},{"no":18174018,"last_modified":1526681502},{"no":18147341,"last_modified":1526680946},{"no":18169269,"last_modified":1526680844},{"no":18175591,"last_modified":1526678867}]},{"page":9,"threads":[{"no":18141720,"last_modified":1526677881},{"no":18172984,"last_modified":1526677184},{"no":18165813,"last_modified":1526677027},{"no":18175584,"last_modified":1526676957},{"no":18156416,"last_modified":1526676925},{"no":18175567,"last_modified":1526676947},{"no":18172296,"last_modified":1526675956},{"no":18145282,"last_modified":1526675707},{"no":18175466,"last_modified":1526674012},{"no":18175467,"last_modified":1526678067},{"no":18163868,"last_modified":1526673219},{"no":18169920,"last_modified":1526668353},{"no":18155214,"last_modified":1526668046},{"no":18164784,"last_modified":1526685130},{"no":18148825,"last_modified":1526663115}]},{"page":10,"threads":[{"no":18152717,"last_modified":1526661825},{"no":18175144,"last_modified":1526706826},{"no":18145901,"last_modified":1526658455},{"no":18155720,"last_modified":1526656459},{"no":18162507,"last_modified":1526656427},{"no":18173804,"last_modified":1526655945},{"no":18172833,"last_modified":1526655602},{"no":18174867,"last_modified":1526650817},{"no":18174798,"last_modified":1526647354},{"no":18164262,"last_modified":1526640740},{"no":18144785,"last_modified":1526629012},{"no":18173863,"last_modified":1526628487},{"no":18158339,"last_modified":1526626796},{"no":18172114,"last_modified":1526622059},{"no":18145594,"last_modified":1526610650}]},{"page":11,"threads":[{"no":18174161,"last_modified":1526665537},{"no":18173783,"last_modified":1526608623}]}]

is what it shows now.

**Comment by alidan** [comment]
Ok, after re hiding everything, only the stickly on /s/ is still hidden.
This is what it looks like after everything unhid itself.
Just as a note, the actual filters are still working perfectly fine.

[{"page":1,"threads":[{"no":12370429,"last_modified":1453430219},{"no":9112225,"last_modified":1798779601},{"no":18151845,"last_modified":1526821053},{"no":18160859,"last_modified":1526820964},{"no":18119520,"last_modified":1526820850},{"no":18172916,"last_modified":1526820703},{"no":18178399,"last_modified":1526821025},{"no":18149575,"last_modified":1526820202},{"no":18175187,"last_modified":1526820152},{"no":18171094,"last_modified":1526820106},{"no":18172382,"last_modified":1526820080},{"no":18178205,"last_modified":1526820066},{"no":18174672,"last_modified":1526819857},{"no":18176198,"last_modified":1526819619},{"no":18160159,"last_modified":1526819142}]},{"page":2,"threads":[{"no":18167850,"last_modified":1526818805},{"no":18178369,"last_modified":1526819135},{"no":18165813,"last_modified":1526818691},{"no":18156698,"last_modified":1526818632},{"no":18154585,"last_modified":1526818377},{"no":18178343,"last_modified":1526818902},{"no":18178213,"last_modified":1526818147},{"no":18165956,"last_modified":1526817765},{"no":18155214,"last_modified":1526817698},{"no":18163958,"last_modified":1526817268},{"no":18147171,"last_modified":1526816829},{"no":18168542,"last_modified":1526816666},{"no":18176641,"last_modified":1526816455},{"no":18178121,"last_modified":1526816288},{"no":18159920,"last_modified":1526816239}]},{"page":3,"threads":[{"no":18178059,"last_modified":1526815946},{"no":18146407,"last_modified":1526815742},{"no":18160360,"last_modified":1526815350},{"no":18164401,"last_modified":1526815239},{"no":18169920,"last_modified":1526814501},{"no":18158657,"last_modified":1526814421},{"no":18160273,"last_modified":1526814005},{"no":18169357,"last_modified":1526813795},{"no":18177226,"last_modified":1526812954},{"no":18166606,"last_modified":1526812319},{"no":18174208,"last_modified":1526811236},{"no":18170532,"last_modified":1526810826},{"no":18173942,"last_modified":1526810497},{"no":18172984,"last_modified":1526809266},{"no":18169657,"last_modified":1526809155}]},{"page":4,"threads":[{"no":18166819,"last_modified":1526808874},{"no":18155720,"last_modified":1526808713},{"no":18169269,"last_modified":1526808504},{"no":18171908,"last_modified":1526808254},{"no":18153487,"last_modified":1526807744},{"no":18178201,"last_modified":1526807392},{"no":18141720,"last_modified":1526807340},{"no":18159979,"last_modified":1526806544},{"no":18173213,"last_modified":1526806521},{"no":18162258,"last_modified":1526806469},{"no":18171550,"last_modified":1526806405},{"no":18164752,"last_modified":1526806401},{"no":18177924,"last_modified":1526806379},{"no":18167437,"last_modified":1526805387},{"no":18177234,"last_modified":1526805261}]},{"page":5,"threads":[{"no":18175591,"last_modified":1526805097},{"no":18177575,"last_modified":1526805005},{"no":18177983,"last_modified":1526804791},{"no":18178133,"last_modified":1526802507},{"no":18176900,"last_modified":1526802130},{"no":18145282,"last_modified":1526801727},{"no":18177947,"last_modified":1526800335},{"no":18174612,"last_modified":1526799468},{"no":18139359,"last_modified":1526798774},{"no":18174521,"last_modified":1526798714},{"no":18152197,"last_modified":1526797690},{"no":18170378,"last_modified":1526797570},{"no":18159709,"last_modified":1526796974},{"no":18178039,"last_modified":1526795968},{"no":18175213,"last_modified":1526795662}]},{"page":6,"threads":[{"no":18176638,"last_modified":1526795179},{"no":18151097,"last_modified":1526794924},{"no":18163893,"last_modified":1526794770},{"no":18171271,"last_modified":1526794743},{"no":18173654,"last_modified":1526794549},{"no":18144762,"last_modified":1526794538},{"no":18157306,"last_modified":1526794341},{"no":18148825,"last_modified":1526793825},{"no":18163404,"last_modified":1526793709},{"no":18148974,"last_modified":1526793432},{"no":18177062,"last_modified":1526793175},{"no":18172854,"last_modified":1526791863},{"no":18171002,"last_modified":1526791702},{"no":18161542,"last_modified":1526791159},{"no":18165315,"last_modified":1526790968}]},{"page":7,"threads":[{"no":18177925,"last_modified":1526819012},{"no":18169769,"last_modified":1526789166},{"no":18173580,"last_modified":1526788900},{"no":18166139,"last_modified":1526788683},{"no":18147784,"last_modified":1526787607},{"no":18176466,"last_modified":1526786400},{"no":18156436,"last_modified":1526786010},{"no":18143725,"last_modified":1526784040},{"no":18152928,"last_modified":1526784021},{"no":18148025,"last_modified":1526784009},{"no":18175987,"last_modified":1526781582},{"no":18163868,"last_modified":1526781584},{"no":18165739,"last_modified":1526781335},{"no":18168719,"last_modified":1526781335},{"no":18167230,"last_modified":1526779383}]},{"page":8,"threads":[{"no":18164272,"last_modified":1526778816},{"no":18160791,"last_modified":1526778141},{"no":18176281,"last_modified":1526776036},{"no":18156727,"last_modified":1526775837},{"no":18177692,"last_modified":1526785000},{"no":18145750,"last_modified":1526773112},{"no":18175760,"last_modified":1526773061},{"no":18156416,"last_modified":1526772608},{"no":18153989,"last_modified":1526771599},{"no":18172943,"last_modified":1526771517},{"no":18158893,"last_modified":1526769924},{"no":18176133,"last_modified":1526768691},{"no":18165725,"last_modified":1526768680},{"no":18148662,"last_modified":1526763316},{"no":18169886,"last_modified":1526763047}]},{"page":9,"threads":[{"no":18175828,"last_modified":1526762491},{"no":18145901,"last_modified":1526761790},{"no":18174582,"last_modified":1526761568},{"no":18169100,"last_modified":1526761469},{"no":18165924,"last_modified":1526761095},{"no":18170838,"last_modified":1526762218},{"no":18176444,"last_modified":1526758547},{"no":18164784,"last_modified":1526754496},{"no":18152717,"last_modified":1526754212},{"no":18174018,"last_modified":1526753747},{"no":18167757,"last_modified":1526747838},{"no":18147341,"last_modified":1526747782},{"no":18174939,"last_modified":1526744275},{"no":18169786,"last_modified":1526743078},{"no":18176193,"last_modified":1526741817}]},{"page":10,"threads":[{"no":18173153,"last_modified":1526741510},{"no":18172296,"last_modified":1526739352},{"no":18176885,"last_modified":1526803032},{"no":18176672,"last_modified":1526769759},{"no":18172833,"last_modified":1526716862},{"no":18166360,"last_modified":1526715470},{"no":18163666,"last_modified":1526713923},{"no":18163238,"last_modified":1526771579},{"no":18170525,"last_modified":1526711386},{"no":18152799,"last_modified":1526700319},{"no":18170093,"last_modified":1526699599},{"no":18158300,"last_modified":1526699239},{"no":18175660,"last_modified":1526694309},{"no":18171782,"last_modified":1526693515},{"no":18174075,"last_modified":1526692521}]},{"page":11,"threads":[{"no":18161468,"last_modified":1526689199},{"no":18175467,"last_modified":1526801134}]}]



on a side note, I had an idea and I posted it https://github.com/ccd0/4chan-x/issues/1933
If there was a way to quick filter only the catalogue view by date, and a quick way to set the date to current time/time of last refresh, that would, at least for me, be a very valuable feature on most boards.

**Comment by ccd0** [comment]
When you hide all the threads on /s/ and it gets unhidden, does that include the stickies?

**Comment by ccd0** [comment]
If the stickies are getting unhidden, can you try this?
Install this test build: https://pomf.pyonpyon.moe/cdbwca.user.js
Ignore any alert windows that pop up until the next step.
Hide the sticky /s/9112225 - questionable age
When it comes unhidden, an alert should pop up. Copy the message from it and paste it here.
Also copy any alert messages from other tabs, if they're different.

**Comment by alidan** [comment]
ok, the questionable age sticky got unhidden at some point, however the other one, the s is not for requests one did stay hidden it seems... 

so I clicked the js and reinstalled, went to /s/ and there was an error, im guessing thats what I was to ignore, and I hid the underage thread alone. 

got a question, is the error/popup up ignoreable/missable at all? 
Also, am I able to hide on other boards and it wont effect what you need from this?

**Comment by ccd0** [comment]
>is the error/popup up ignoreable/missable at all?

The only way you could miss it is by closing the tab that it showed up in without checking it first.

>Also, am I able to hide on other boards and it wont effect what you need from this?

It shouldn't make any difference.

**Comment by ccd0** [comment]
Did it update to v1.14.1.2? That would explain the lack of message.

Can you see if you can trigger the issue by installing a version of the script over the old one?

And if that doesn't work, can you try again with
https://pomf.pyonpyon.moe/wagxji.user.js
which should have auto-updating disabled?

By the way, what userscript manager are you using?

**Comment by ccd0** [comment]
Also, please check whether you might have multiple copies of 4chan X enabled somehow. If different copies are loading at different times, it could explain your problem.

**Comment by alidan** [comment]
tampermonkey
I used the 4chan extension till I tried going to the non extension as someone said it took longer for it to wipe anything, the chrome extension is disabled. 
only 1 4chanx is in the tampermonkey

not quite sure how to get an old version, then update it to see if that triggered anything, I couldn't make tampermonkey update the '1.14.0.9' version to something newer to see if it did anything to the page. 
I do have another issue that may be related, every now and then this happened https://i.imgur.com/PEgK2Eo.png im not 100% sure what the cause is. 
going to try again with the the linked script

**Comment by alidan** [comment]
day 8
None of the boards I have visited reverted yet
/s/ only has the stickies and 2 other threads hidden, not sure when I hid them
everything else, is culled through and mass hidden. 
I'm starting to think there may be an issue with going from an older version to the newer one that may be a problem, going to continue this for at least a month possibly longer if I forget i'm on a version that's not auto updating.

**Comment by alidan** [comment]
/h/ and /d/ have forgotten their hides
a few posts where forgotten too
The only things filtered are manual filters. 

looking at my hidden number, it was at 1500~ its now at 1200~

**Comment by alidan** [comment]
Ok got the error the questionable age sticky on /s/ was forgotten that it was hidden

https://i.imgur.com/E6b5T4n.png
https://i.imgur.com/X0jqWOH.png
https://i.imgur.com/lOcljyd.png
https://i.imgur.com/1gvLzlg.png

I wasn't able to copy it, but I should have screenshotted the whole thing. 
for reference sake I am on 1.14.1.1 that you posted a few up, so its not a I updated the script issue. 
I also just found out why 2 other threads were filtered, I had a catch all filter of /r*ddit/ due to how annoying people were getting and it caught 2 threads it shouldnt have, after killing that filter I got another error with all the threads no longer filtered

https://i.imgur.com/EziRYKC.png
https://i.imgur.com/BKMDPhF.png
https://i.imgur.com/Lf0zwWg.png
https://i.imgur.com/DYz9Vxr.png

hope this helps, anything else I should do?

**Comment by alidan** [comment]
also, its now erroring on every single hide across all boards apparently

**Comment by ccd0** [comment]
Was this the first error message? If so, this is an issue with Tampermonkey and not something I can fix from 4chan X's end.

**Comment by ccd0** [comment]
>also, its now erroring on every single hide across all boards apparently

This is expected and will keep happening until the sticky is hidden again.

**Comment by ccd0** [comment]
I'm going to try backing out the workaround for #1323 (7b8c2df5e4aae96b47771c0bb90989765d719d5c) which should no longer be relevant.

**Comment by alidan** [comment]
Ok, im trying to get violentmonkey to work to see if the issue is there too, I got every other extension working with it but 4chanx any ideas?

as far as I can tell, the extension and the script at least with tampermonkey are having the same issue with forgetting things, if its possible tampermonkey was the problem, it worth seeing if it presents itself in violentmonkey.

**Comment by ccd0** [comment]
Just pushed a new beta version (v1.14.2.0) which might solve the problem.

**Comment by ccd0** [comment]
If I'm right about what's causing it this time, this was caused by improper handling of occasional I/O errors. Since these could be originating from the browser or even from the OS itself, the choice of Tampermonkey vs Violentmonkey might not matter here.

**Comment by alidan** [comment]
Ok, going to turn tampermonkey on then, the issue was ublock was not letting the script load, and I was getting nowhere without whitelisting 4chan which I can't bring myself to do due to the sketchy ads

**Comment by saxamaphone69** [comment]
FWIW, I've been having stickies randomly/periodically unhide themselves over the last 2-3 weeks (I feel like it coincided with when the "new" version of the bottom ads appeared) , and have been keeping an eye on this issue.

**Comment by alidan** [comment]
new version almost immediately forgot every single thing that was hidden on 8 of the 9 boards that I hide things on, however everything in a normal filter still catches everything. 

Honestly speaking, after screwing around with creation filtering, bump filtering feels kind of useless, I mean either I watch what I want to keep up with, or the thread served its purpose, if you could make a creation time filter, have an enable it button in options, and then an update button to the left of the sort methods, this would solve most of my problems. I only see what's new, I watch what I want to keep up with, and the rest is filtered and stored in the user defined method, as that one seems to not lose its filters at all.

as for violent monkey, anywhere I stick 
@@||boards.4chan.org^$csp
It doesn't do anything, along with every permutation I can think of and in every field I could stick it in so I can't test with that to see if it works.

**Comment by alidan** [comment]
Ok, its been nearly a day since my last post. 

I just went to every board where I hid shit and did a tally, 
158 threads are hidden, either manually hidden or through a filter
however 4chanx still claims 831
https://i.imgur.com/Xd7a0u8.png


just looking at all my boards I hide everything on, the number should come to 1350, I saw that number often enough when everything is hidden properly, however, it seems like things are still in memory but are not being hidden. 

im going to clear the hide after exporting and re hide things. any thoughts on what is going on with this?

**Comment by ccd0** [comment]
The number may include archived threads. You can check exactly which threads by looking at your exported settings.

**Comment by gotmilk0112** [comment]
It's having a bit of a different issue, now. I'll wake up in the morning and manually hide a few threads, and then a little while later, only those few threads that I hid, will un-hide again. But only once; they stay hidden the rest of the day.

This happens once every 2-3 days.

Running version 1.14.1.2

**Comment by ccd0** [comment]
Can you post your exported settings?

**Comment by alidan** [comment]
[New WinRAR ZIP archive.zip](https://github.com/ccd0/4chan-x/files/2087229/New.WinRAR.ZIP.archive.zip)

Have no idea if this worked, it should have 2016 hidden, have no idea where a little less then half of them came from because the boards I hide dont have the turn around to in 2 days have 700 new threads to hide.

**Comment by ccd0** [comment]
Yeah, something is odd there. I see threads that were archived as far back as May 28. Are you positive you cleared the list of hidden threads on June 6? Did you use the button in the settings to clear the list? Have you been using incognito mode and/or the native catalog during any of this?

**Comment by alidan** [comment]
I exported everything before clearing it, then hid everything and went on, im nearly 100% positive it was clear and something happened in the meantime. 

I wasn't really keeping a daily look at it, I only noticed it when I was looking for a setting to do something else. 

would it be possible for me to clear the hidden threads in one tab, not closing other ones and having the hidden tabs that should have been cleared re construct themselves based on other open tabs? that's really the only way I can see them coming back through a fault of my own.

**Comment by ccd0** [comment]
Have you been using incognito mode and/or the native catalog during any of this?

**Comment by alidan** [comment]
in the last since cleared the hidden, I shouldn't have.
settings with the catalog have likely never been changed

**Comment by ccd0** [comment]
>would it be possible for me to clear the hidden threads in one tab, not closing other ones and having the hidden tabs that should have been cleared re construct themselves based on other open tabs?

It could happen if you hid/unhid threads in one of those tabs after clearing the list.

**Comment by alidan** [comment]
nah, the tabs that were left open were just threads that I have yet to read through, nothing about hiding when to there.

**Comment by alidan** [comment]
Ok, going to do it again now that it has forgotten every single one of the hidden threads
checking the scripts page, it says 800 tabs are hidden

[21 hidden things accross d h gif aco e trash s hc hr.zip](https://github.com/ccd0/4chan-x/files/2093348/21.hidden.things.accross.d.h.gif.aco.e.trash.s.hc.hr.zip)

however only 21 are hidden on any of the boards i go to and hide things. 
This is a number that seems to pop up a bit too, are you able to see anything weird with it?
It honestly seems like its 2 files that are conflicting with each other as to which one is going to be used and it keeps going back to a really old file for some reason.

going to clear it and make sure I don't use incognito for a while.

**Comment by alidan** [comment]
just a note, I cleared the hidden, it went to 0, I reloaded to make sure, and it was still 800
I cleared it again, and refreshed several times, the 800 was no longer there. I then went and refreshed every page, and I noticed something

filter hides aren't tracked as hidden

I have to ask, is there a functional difference between hides and filters? It seems filters are never forgotten but hides are kind of routinely forgotten now.
would a stop gap of rolling page hides into user filters work? it would need to be manually culled every so often for performance sure, but till its able to work automatically without issue.

On a side note I have been trying to figure out if there is a way in script to filter dates/time

let's give an example, I cull through boards to see what threads I want to see and what ones I dont, 
lets use hr as an example, as of posting this, https://boards.4chan.org/hr/thread/3172450 is the most recent thread, and 06/11/18(Mon)9:27:55 pm is the time, 

If I culled through the thread now, it would be 6/12/18(Tue)4:31:55 am
Is there a way to apply this to a general filter that would filter out anything older than the time I input with another filter for boards:d,h,gif,aco,e,trash,s,hc,hr;
while it wouldn't be ideal for what I do, and having one set up for those boards and then possibly other ones would be a pain in the ass to keep up with manually, it would at least not constantly revert like it is now for reasons i'm not able to identify.

**Comment by ccd0** [comment]
>Ok, going to do it again now that it has forgotten every single one of the hidden threads

Is there any possibility these thread hides could be simply not be getting saved in the first place? Was any thread hide lost that was at one point there when the page was loaded/reloaded?

>It honestly seems like its 2 files that are conflicting with each other as to which one is going to be used and it keeps going back to a really old file for some reason.

It might be worth double-checking again that you do not have two versions of 4chan X enabled anywhere, including the extension, Tampermonkey scripts, Violentmonkey scripts, etc.

>just a note, I cleared the hidden, it went to 0, I reloaded to make sure, and it was still 800

Is it possible to hide a bunch of threads and reproduce this again? If so, it would be useful to check if an error was logged in the console when you cleared the threads.

>would a stop gap of rolling page hides into user filters work?

If there's a data storage bug, we have to fix it. It's quite possible that you would notice your filters getting reverted if you changed them more often.

**Comment by saxamaphone69** [comment]
Personally, my filters are still working with no issues (but I only have highlight filters not hiding ones).

I have not been to 4chan in Incognito but my threads still find themselves unhidden.

My settings says `Hidden: 0`, I hide the sticky on /fit/ (which found itself unhidden), refresh the page, and it still says `Hidden: 0` but the sticky is now hidden.

I had console open this entire time and no errors or warnings appeared.

**Comment by alidan** [comment]
everything that would be a second version has been disabled, I can outright remove everything that would be a second version, ill do that when/if it reverts again. 

as of right now everything is currently hidden, however I'm also not getting and status on how much is hidden and it says 0 for me too.

**Comment by alidan** [comment]
Ok sometime in the last two days a few of the boards forgot their hides, going to go about removing every other disabled version I have
and just as a note, hidden still reads as zero

**Comment by bakugo** [comment]
Found the issue (I think)

It's actually not related to storage at all, it's caused by [DataBoard's ajaxClean function](https://github.com/ccd0/4chan-x/blob/6e2c93e09e2704d14f96b580909651a881429cae/src/classes/DataBoard.coffee#L124). Occasionally, when the second callback (for archive.json) is executed, the response of the first request (for threads.json) `e1.target.response` is null and this gets passed to ajaxCleanParse which clears every active thread as a result. I have no idea what causes the response to become null over time (this might be worth investigating) but simply storing it in a variable during the first callback should fix it. I'm using Chrome 67, in case this behavior is browser specific.

The Hidden button in the settings is also broken due to another problem: it was not updated for multisite support. The hiddenThreads and hiddenPosts storage keys can contain data split by site, but the button doesn't handle that, it just tries to read hiddenThreads.boards directly when it can be at hiddenThreads["4chan.org"].boards instead.

**Comment by ccd0** [comment]
The button count was fixed by 23c764347bb68a9b09d83197c43959c3996abb54, so closing.
